src/moveit_custom_api.cpp

```diff
@@ -197,59 +255,65 @@ void MoveitCustomApi::addCollissionObjects()
   primitive.type = primitive.BOX;
   primitive.dimensions.resize(3);
   primitive.dimensions[0] = TOTAL_INNER_CELL_X_DIMENSION_;
-  primitive.dimensions[1] = TOTAL_INNER_CELL_Y_DIMENSION_;
+  primitive.dimensions[1] = std::abs(TOTAL_INNER_CELL_Y_DIMENSION_);
   primitive.dimensions[2] = 0;
 
   // Define a pose for the box (specified relative to frame_id)
   geometry_msgs::Pose box_pose;
   box_pose.orientation.w = 1.0;
-  box_pose.position.x = -(BASE_OFFSET_FROM_LEFT_WALL_-BASE_OFFSET_FROM_RIGHT_WALL_)/2;  // Not perfectly symmetrical.
-  box_pose.position.y = TOTAL_INNER_CELL_Y_DIMENSION_/2-BASE_OFFSET_FROM_BACK_WALL_; // Base is ofset by (0.1470/2-.275)
-  box_pose.position.z = -0.01; //Push it slightly down to avoid collission with base plate.
+  box_pose.position.x = -(BASE_OFFSET_FROM_LEFT_WALL_ - BASE_OFFSET_FROM_RIGHT_WALL_) / 2; // Not perfectly symmetrical.
+  box_pose.position.y = TOTAL_INNER_CELL_Y_DIMENSION_ / 2 - BASE_OFFSET_FROM_BACK_WALL_;   // Base is ofset by (0.1470/2-.275)
+  if (robot == "ur_ros" || robot == "ur")
```

Comment:
this should also be parametrised outside in the config files. The idea is that the custom API should be usable with any robot, not just UR or PRBT.